pulp_python/app/serializers.py

```diff
@@ -237,11 +237,12 @@ def deferred_validate(self, data):
                 # Copy file to a temp directory under the user provided filename, we do this
                 # because pkginfo validates that the filename has a valid extension before
                 # reading it
-                with tempfile.TemporaryDirectory() as td:
-                    temp_path = os.path.join(td, filename)
+                with tempfile.NamedTemporaryFile('wb', suffix=filename) as temp_file:
                     artifact = data["artifact"]
-                    shutil.copy2(artifact.file.path, temp_path)
-                    metadata = DIST_TYPES[packagetype](temp_path)
+                    artifact_file = storage.open(artifact.file.name)
+                    shutil.copyfileobj(artifact_file, temp_file)
```

Comment:
Nice, I didn't know about that function.

Reply:
Credit goes to @fao89!